Packs/IronDefense/Integrations/IronDefense/IronDefense.py

```diff
@@ -241,14 +406,236 @@ def report_observed_bad_activity(self, name, description='', ip='', domain='',
+            self.logger.error('Failed to retrieve alerts. The response failed with status code {}. The response was: {}'
+                              .format(response.status_code, err_msg))
+            raise HTTPException('Failed to retrieve alerts ({}): {}'
+                                .format(response.status_code, err_msg))
+
+    def get_alert_irondome_information(self, alert_id):
+        self.logger.debug('Retrieving Alert IronDome Information: Alert ID={}'.format(alert_id))
+
+        req_body = {
+            'alert_id': alert_id,
+        }
+        response = self._http_request('POST', '/GetAlertIronDomeInformation', body=json.dumps(req_body))
+        if response.status_code != 200:
+            err_msg = self._get_error_msg_from_response(response)
+            self.logger.error('Failed to retrieve IronDome information for alert with ID ({}). The response failed '
+                              'with status code {}. The response was: {}'.format(alert_id, response.status_code,
+                                                                                 response.text))
+            raise HTTPException('Failed to retrieve IronDome information for alert with ID {} ({}): {}'
+                                .format(alert_id, response.status_code, err_msg))
+        else:
+            self.logger.debug('Successfully retrieved IronDome information for alert ({})'.format(alert_id))
+            dome_alert_info = response.json()
+            return dome_alert_info
+
 
 ''' COMMANDS MANAGER / SWITCH PANEL '''
 
 
-# This will be enabled at a later date.
-# def fetch_incidents_command():
-#     incidents = IRON_DEFENSE.fetch_incidents()
-#     demisto.incidents(incidents)
+def fetch_incidents_command():
+    # IronDome Notification related params
+    dome_categories = PARAMS.get('domeCategories', None)
+    dome_limit = int(PARAMS.get('domeLimit', 200))
+    disable_dome_notifs = not PARAMS.get('enableDomeNotifications', False)
+    # Alert Notification related params
+    alert_categories = PARAMS.get('alertCategories', None)
+    alert_subcategories = PARAMS.get('alertSubCategories', None)
+    alert_severity_lower = PARAMS.get('alertSeverityLower', None)
+    alert_severity_upper = PARAMS.get('alertSeverityUpper', None)
+    alert_limit = int(PARAMS.get('alertLimit', 200))
+    alert_actions = PARAMS.get('alertActions', None)
+    disable_alert_notifs = not PARAMS.get('enableAlertNotifications', True)
+    # Event Notification related params
+    event_categories = PARAMS.get('eventCategories', None)
+    event_subcategories = PARAMS.get('eventSubCategories', None)
+    event_severity_lower = PARAMS.get('eventSeverityLower', None)
+    event_severity_upper = PARAMS.get('eventSeverityUpper', None)
+    event_limit = int(PARAMS.get('eventLimit', 200))
+    event_actions = PARAMS.get('eventActions', None)
+    disable_event_notifs = not PARAMS.get('enableEventNotifications', False)
+
+    incidents: list = []
```

Comment:
General question about the fetch, don't you need a time marker to fetch from?

Reply:
No our fetch is done off a message queue so the first notification fetched would next unsent message on the queue.